packages/aws-rfdk/lib/deadline/lib/sep-spotfleet.ts

```diff
@@ -0,0 +1,693 @@
+   * have AWSThinkboxDeadlineSpotEventPluginWorkerPolicy policy attached and
+   * the role name must begin with "DeadlineSpot":
+   *
+   *    const role = new iam.Role(this, 'MyRole', {
+   *      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+   *      managedPolicies: [
+   *        ManagedPolicy.fromAwsManagedPolicyName('AWSThinkboxDeadlineSpotEventPluginWorkerPolicy'),
+   *      ],
+   *      roleName: 'DeadlineSpot' + ...,
+   *    });
+   *
+   * @default - A role will automatically be created.
+   */
+  readonly role?: IRole;
+
+  /**
+   * Name of SSH keypair to grant access to instances.
+   *
+   * @default - No SSH access will be possible.
+   */
+  readonly keyName?: string;
+
+  /**
+   * Security Groups to assign to this fleet.
+   *
+   * @default - A new security group will be created automatically.
+   */
+  readonly securityGroups?: ISecurityGroup[];
+
+  /**
+   * User data that instances use when starting up.
+   *
+   * @default - User data will be created automatically.
+   */
+  readonly userData?: UserData;
+
+  /**
+   * The Block devices that will be attached to your workers.
+   *
+   * @default - The default devices of the provided ami will be used.
+   */
+  readonly blockDevices?: BlockDevice[];
+
+  /**
+   * Indicates how to allocate the target Spot Instance capacity
+   * across the Spot Instance pools specified by the Spot Fleet request.
+   *
+   * @default - SEPSpotFleetAllocationStrategy.LOWEST_PRICE.
+   */
+  readonly allocationStrategy?: SEPSpotFleetAllocationStrategy;
+
+  /**
+   * Where to place the instance within the VPC.
+   *
+   * @default - Private subnets.
+   */
+  readonly vpcSubnets?: SubnetSelection;
+
+  /**
+   * The end date and time of the request, in UTC format (YYYY -MM -DD T*HH* :MM :SS Z).
+   * After the end date and time, no new Spot Instance requests are placed or able to fulfill the request.
+   *
+   * @default - the Spot Fleet request remains until you cancel it.
+   */
+  readonly validUntil?: Expiration;
+
+  /**
+   * Properties for setting up the Deadline Worker's LogGroup
+   * @default - LogGroup will be created with all properties' default values and a prefix of "/renderfarm/".
+   */
+  readonly logGroupProps?: LogGroupFactoryProps;
+
+  /**
+   * An optional provider of user data commands to be injected at various points during the Worker configuration lifecycle.
+   * You can provide a subclass of InstanceUserDataProvider with the methods overridden as desired.
+   */
+  readonly userDataProvider?: IInstanceUserDataProvider;
+}
+
+/**
+ * Synthesize an array of block device mappings from a list of block device
+ *
+ * @param construct the instance/asg construct, used to host any warning
+ * @param blockDevices list of block devices
+ */
+function synthesizeBlockDeviceMappings(construct: Construct, blockDevices: BlockDevice[]): CfnLaunchConfiguration.BlockDeviceMappingProperty[] {
```

Comment:
I left it here intentionally for reviewers, but forgot to add my own comment first.
This implementation was copied from the CDK code, where it is private and can't be imported.
I wanted to put this function in a separate file, but it can't be exported as is, because we get 
`JSII9998: Unsupported FunctionDeclaration node. This declaration will not be accessible from other languages.`.

So I will probably just make it a private member function, it seems that both reviewers suggested that.

Reply:
This has been moved to another class with the rest of the code responsible for construction of the Spot Fleet Request Configurations.